pre_processing/read_aatsr_dimensions.F90

```diff
@@ -80,7 +83,7 @@ end subroutine get_aatsr_dimension
    logical,                        intent(in)  :: verbose
 
    character(len=path_length,kind=c_char) :: l1b_file_C
-   integer(c_short)                       :: tmp_dynght, half_orbit
+   integer(c_short)                       :: tmp_dynght, half_orbit, c_dynght
```

Comment:
I don't think you needed to make an extra variable here, as `tmp_dynght` is only passed to C routines or considered in if statements.

Reply:
I just wanted to avoid any mismatch between integer(kind=sint) and integer(c_short). Can we be sure that they will always be the same?

Reply:
Well that's already happening at https://github.com/ORAC-CC/orac/blob/2111284a33fe44ea6ae26d38def159c84682e968/pre_processing/read_aatsr_dimensions.F90#L102

You could explicitly cast to `c_short` on that line and do comparisons against the original variable

Reply:
It's quite possible I don't really understand how Fortran does the type casting, but I was working on the assumption that you can move the value between numeric types using the `=` operator, but should always ensure the type of variables passed in and out of sub-routines matches with what is defined in the sub-routine?
`day_night` is passed in as a Fortran `integer(kind=sint)` but is then passed on to the C function as a `integer(c_short)` . Could passing `day_night` directly to the C function not be broken if the definition of `sint` changed (or indeed, if the definition of `c_short` was different on some future system?